sogs/routes/rooms.py

```diff
@@ -960,3 +960,27 @@ def delete_all_posts(room, sid):
     if not deleted:
         abort(http.NOT_FOUND)
     return jsonify({})
+
+@rooms.delete("/room/all/<SessionID:sid>")
+def delete_user_posts_from_all_rooms(sid):
+    """
+    Deletes all posts made by a user from all rooms
+
+    # URL Parameters
+
+    - `sid` — the session id of the user to ban
+
+    # Return value
+
+    An empty json object is returned.
+
+    # Error status codes
+
+    - 403 Forbidden — if the invoking user does not have access to a room.
+    """
+    user = muser.User(session_id=sid, autovivify=False)
+    for room in mroom.get_accessible_rooms(g.user):
+        deleted, _ = room.delete_all_posts(user, deleter=g.user)
+        if not deleted:
+            app.logger.warning(f"Error deleting posts from room: {room}.")
```

Comment:
`deleted = 0` isn't really a failure case; it can just mean the user had no posts in that room, which would happen pretty often (since it's rare that someone spams in *every single* room).

Rather this call throws for permission errors (e.g. if you aren't a moderator of the room, or aren't an admin and are trying to delete an admin's posts), but in such a case we likely want to continue trying with other rooms, so something like:
```Python
for room in mroom.get_accessible_rooms(g.user):
    try:
        room.delete_all_posts(user, deleter=g.user)
    except BadPermission:
        pass
```

Reply:
Much better. Thanks.